keanu-project/src/main/java/io/improbable/keanu/model/LinearRegression.java

```diff
@@ -0,0 +1,85 @@
+package io.improbable.keanu.model;
+
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+public class LinearRegression implements LinearModel {
+
+    private static final double DEFAULT_PRIOR_ON_WEIGHTS = 2.0;
+    private static final double MU_OF_WEIGHTS = 0.0;
+    private static final VertexLabel WEIGHTS_LABEL = new VertexLabel("weights");
+    private static final VertexLabel INTERCEPT_LABEL = new VertexLabel("intercept");
+    private static final VertexLabel Y_OBSERVATION_LABEL = new VertexLabel("y");
+
+    private DoubleTensor x;
+    private DoubleTensor y;
+    private BayesianNetwork net;
+    private boolean isFit = false;
+    private double sigmaOnPrior;
+
+    public LinearRegression(DoubleTensor x, DoubleTensor y) {
+        this(x, y, DEFAULT_PRIOR_ON_WEIGHTS);
+    }
+
+    public LinearRegression(DoubleTensor x, DoubleTensor y, double sigmaOnPrior) {
+        this.x = x;
+        this.y = y;
+        this.sigmaOnPrior = sigmaOnPrior;
+        construct();
+    }
+
+    @Override
+    public BayesianNetwork construct() {
+        int numberOfFeatures = x.getShape()[0];
+        int[] shapeOfWeights = new int[]{1, numberOfFeatures};
+        DoubleVertex weights = new GaussianVertex(shapeOfWeights, MU_OF_WEIGHTS, sigmaOnPrior).setLabel(WEIGHTS_LABEL);
+        DoubleVertex intercept = new GaussianVertex(MU_OF_WEIGHTS, sigmaOnPrior).setLabel(INTERCEPT_LABEL);
+        DoubleVertex xMu = weights.getValue().isScalar() ? weights.times(ConstantVertex.of(x)) : weights.matrixMultiply(ConstantVertex.of(x));
```

Comment:
aside: It would be nice if `matrixMultiply` just did the right thing for scalars and you didn't need the branch here.

This pattern comes up in other places too (e.g. MultivariateGaussian)

Reply:
Agreed - let me think about this. Possibly a separate task.